common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -612,6 +603,39 @@ protected void validateTokenResponse(@NonNull final MicrosoftStsTokenRequest req
         validateTokensAreInResponse(request, response);
     }
 
+    public TokenResult requestToken(
```

Comment:
We have a method with same name but slightly different signature in the parent class `OAuth2Strategy`. Why not just refactor that method and change the signature (rather than creating an almost-duplicate here)?

Reply:
Added requestToken() at OAuth2Strategy level itself with a new ITokenResponseHandler interface.